zhaquirks/tuya/__init__.py

```diff
@@ -226,6 +227,31 @@ def payload(self, value):
         else:
             raise ValueError(f"Unknown {self.dp_type} datapoint type")
 
+    def __new__(cls, *args, **kwargs):
+        """Disable copy constrctor."""
+        return super().__new__(cls)
+
+    def __init__(self, value=None, function=0, *args, **kwargs):
+        """Convert from a zigpy typed value to a tuya data payload."""
+        self.function = function
+
+        if value is None:
+            return
+        elif isinstance(value, (t.bitmap8, t.bitmap16, t.bitmap32)):
+            self.dp_type = TuyaDPType.BITMAP
+        elif isinstance(value, (bool, t.Bool)):
+            self.dp_type = TuyaDPType.BOOL
+        elif isinstance(value, enum.Enum):
+            self.dp_type = TuyaDPType.ENUM
+        elif isinstance(value, int):
+            self.dp_type = TuyaDPType.VALUE
+        elif isinstance(value, str):
+            self.dp_type = TuyaDPType.STRING
+        elif isinstance(value, t.Struct):
+            self.dp_type = TuyaDPType.RAW
+
```

Comment:
Maybe a safe check here to be sure to assign the `dp_type`?

Reply:
I have a different idea, to set it default to RAW

Reply:
Done